src/core/communication.cpp

```diff
@@ -392,9 +392,6 @@ void mpi_gather_stats(int job, void *result, void *result_t, void *result_nb,
                   (double *)result_t_nb, 1);
     break;
   case 4:
```

Comment:
I think you should also remove the cases. It's an error now to call it with 'job = 4`.